src/main/c/NativeLongFunc.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// src/main/c/NativeFunc.cpp - native funcs for Scala NativeLongFunc
+//
+// Richard Cownie, Hail Team, 2018-04-18
+//
+#include "hail/NativePtr.h"
+#include "hail/CommonDefs.h"
+#include <assert.h>
+#include <jni.h>
+
+NAMESPACE_BEGIN(hail)
+
+NAMESPACE_BEGIN_ANON
+
+typedef NativeFuncObj<long> NativeFunc;
+
+NativeFunc* toNativeFunc(JNIEnv* env, jobject thisObj) {
+  // Should be a dynamic_cast, but RTTI causes trouble
+  return reinterpret_cast<NativeFunc*>(getFromNativePtr(env, thisObj));
+}
+
+NAMESPACE_END_ANON
+
+NATIVEMETHOD(long, NativeLongFuncL0, apply)(
```

Comment:
Can you set up a brief meeting where you explain what this stuff is doing to the group?